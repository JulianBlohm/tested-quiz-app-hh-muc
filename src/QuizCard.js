import React, { useState } from 'react'
import styled from 'styled-components/macro'

export default function QuizCard({ question, answer }) {
  const [showAnswer, setShowAnswer] = useState(false)

  return (
    <Wrapper>
      <p>{question}</p>
      <button onClick={() => setShowAnswer(!showAnswer)}>Toggle answer</button>
      {showAnswer && <p>{answer}</p>}
    </Wrapper>
  )
}

const Wrapper = styled.section`
  margin: 10px 10px 40px;
  padding: 20px;
  box-shadow: 0 10px 16px #0004;
`
