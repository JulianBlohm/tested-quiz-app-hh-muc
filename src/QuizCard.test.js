import QuizCard from './QuizCard'
import { render, fireEvent } from '@testing-library/react'

describe('QuizCard', () => {
  it('renders correctly', () => {
    const { container } = render(<QuizCard question="Foo?" answer="Bar!" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('shows the question and answer texts', () => {
    const { getByText, queryByText, getByRole } = render(
      <QuizCard question="Foo?" answer="Bar!" />
    )
    expect(getByText(/foo/i)).toBeInTheDocument()
    expect(queryByText(/bar/i)).not.toBeInTheDocument()

    const button = getByRole('button', { name: 'Toggle answer' })
    fireEvent.click(button)

    expect(queryByText(/bar/i)).toBeInTheDocument()
  })
})
