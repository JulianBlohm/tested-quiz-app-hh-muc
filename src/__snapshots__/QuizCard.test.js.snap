// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizCard renders correctly 1`] = `
.c0 {
  margin: 10px 10px 40px;
  padding: 20px;
  box-shadow: 0 10px 16px #0004;
}

<section
  class="c0"
>
  <p>
    Foo?
  </p>
  <button>
    Toggle answer
  </button>
</section>
`;
